packages/@aws-cdk/aws-cloudwatch/lib/dashboard.ts

```diff
@@ -108,6 +113,8 @@ export class Dashboard extends Resource {
       }),
     });
 
+    this.dashboardName = dashboard.dashboardName;
```

Comment:
Since the dashboard name is the resource `physicalName`  you need to use `getResourceNameAttribute` . 
This will ensure that if a name is not passed the `Ref` value will be used (and resolved at deploy time).

Take a look at examples such as S3 `bucketName`, autoscaling `autoScalingGroupName`